flink-core/src/main/java/org/apache/flink/api/common/state/StateTtlConfig.java

```diff
@@ -360,6 +373,8 @@ public StateTtlConfig build() {
 
 		private static final CleanupStrategy EMPTY_STRATEGY = new EmptyCleanupStrategy();
 
+		private boolean isCleanupInBackground = false;
```

Comment:
Let's make it final. If we move `isCleanupInBackground` and `strategies` to builder, we can make `CleanupStrategies` immutable by passing `isCleanupInBackground` and `strategies` to `CleanupStrategies` constructor in `builder.build()`. `CleanupStrategies` field can be removed from builder then. `activate` and `setCleanupInBackground` are also not needed in `CleanupStrategies`.

Reply:
OK, but the `EMPTY_STRATEGY` must be public because it must be used in `cleanupFullSnapshot`. If it is not public, there is a grammar error.